### ISSUE TITLE
[e2e] a11y reports are no longer being written to disk

### ISSUE THREAD

**Issue by unlikelyzero** [nasa/openmct#7317]
#### Summary
After merging #7047 , I received a PR comment from @scottbell  to 

> i think we should be using `fs/promises` here instead. Also, we shouldn't check for the file's existence before creating the directory as we could get into a [race condition](https://nodejs.org/docs/latest-v18.x/api/fs.html#fs_fs_exists_path_callback). something like this instead:
> 
> ```js
> const fs = require('fs').promises;
> 
> async function writeAccessibilityReport(reportPath, accessibilityScanResults) {
>     try {
>         try {
>             await fs.mkdir(TEST_RESULTS_DIR, { recursive: true });
>         } catch (error) {
>             if (error.code !== 'EEXIST') {
>                 throw error; // Throw the error if it is not because the directory already exists
>             }
>         }
>         
>         const data = JSON.stringify(accessibilityScanResults, null, 2);
>         await fs.writeFile(reportPath, data); 
>         console.log(`Accessibility report with violations saved successfully as ${reportPath}`);
>         return accessibilityScanResults;
>     } catch (err) {
>         console.error(`Error writing the accessibility report to file ${reportPath}:`, err);
>         throw err;
>     }
> }
> ```

This ticket tracks that work
